src/utils/calculateBasicAuthUsernamePasswordPair.js

```diff
@@ -0,0 +1,32 @@
+import { oauth2 } from './oauth2'
+
+export function calculateBasicAuthUsernamePasswordPair ({
+  username,
+  password,
+  token,
+  oauth2format
+}) {
+  const key = [!!username, !!password, !!token, !!oauth2format]
+    .map(Number)
+    .join('')
+  // See helpful truth table in documentation
+  // prettier-ignore
+  switch (key) {
+    case '0000': return null
+    case '1000': throw new Error(`Missing password or token`)
+    case '0100': throw new Error(`Missing username`)
+    case '1100': return { username, password }
+    case '0010': return { username: token, password: '' }
+    case '1010': return { username, password: token }
+    case '0110': throw new Error(`Missing username`)
```

Comment:
Is this a good error? If I go fix this and add a username, I will end up with a new error - `1110`.

Reply:
I know right?! That drove me nuts, but I couldn't think of a good alternative at the time. If I made it the same error as 1110, then when they remove either password or token... oh hang on. Removing password would actually be valid because of the Github shorthand.